crates/trigger-http/src/lib.rs

```diff
@@ -1000,3 +1007,212 @@ mod tests {
+                tracing::warn!("tls protocol error: {e:?}");
+                types::ErrorCode::TlsProtocolError
+            })?;
+            let stream = TokioIo::new(stream);
+
+            let (sender, conn) = timeout(
+                connect_timeout,
+                hyper::client::conn::http1::handshake(stream),
+            )
+            .await
+            .map_err(|_| types::ErrorCode::ConnectionTimeout)?
+            .map_err(hyper_request_error)?;
+
+            let worker = wasmtime_wasi::runtime::spawn(async move {
+                match conn.await {
+                    Ok(()) => {}
+                    // TODO: shouldn't throw away this error and ideally should
+                    // surface somewhere.
+                    Err(e) => tracing::warn!("dropping error {e}"),
+                }
+            });
+
+            (sender, worker)
+        }
+    } else {
+        let tcp_stream = TokioIo::new(tcp_stream);
+        let (sender, conn) = timeout(
+            connect_timeout,
+            // TODO: we should plumb the builder through the http context, and use it here
+            hyper::client::conn::http1::handshake(tcp_stream),
+        )
+        .await
+        .map_err(|_| types::ErrorCode::ConnectionTimeout)?
+        .map_err(hyper_request_error)?;
+
+        let worker = wasmtime_wasi::runtime::spawn(async move {
+            match conn.await {
+                Ok(()) => {}
+                // TODO: same as above, shouldn't throw this error away.
+                Err(e) => tracing::warn!("dropping error {e}"),
+            }
+        });
+
+        (sender, worker)
+    };
+
+    // at this point, the request contains the scheme and the authority, but
+    // the http packet should only include those if addressing a proxy, so
+    // remove them here, since SendRequest::send_request does not do it for us
+    *request.uri_mut() = http::Uri::builder()
+        .path_and_query(
+            request
+                .uri()
+                .path_and_query()
+                .map(|p| p.as_str())
+                .unwrap_or("/"),
+        )
+        .build()
+        .expect("comes from valid request");
+
+    let resp = timeout(first_byte_timeout, sender.send_request(request))
+        .await
+        .map_err(|_| types::ErrorCode::ConnectionReadTimeout)?
+        .map_err(hyper_request_error)?
+        .map(|body| body.map_err(hyper_request_error).boxed());
+
+    Ok(wasmtime_wasi_http::types::IncomingResponse {
+        resp,
+        worker: Some(worker),
+        between_bytes_timeout,
+    })
+}
+
+fn get_client_tls_config_for_authority(
+    authority: &String,
+    client_tls_opts: Option<HashMap<String, ParsedClientTlsOpts>>,
+) -> rustls::ClientConfig {
+    // derived from https://github.com/tokio-rs/tls/blob/master/tokio-rustls/examples/client/src/main.rs
+    let mut root_cert_store = rustls::RootCertStore {
+        roots: webpki_roots::TLS_SERVER_ROOTS.into(),
```

Comment:
If you're providing your own certs, you probably don't want to also trust standard roots? (or they should at least be disable-able)

Reply:
This impl is copied from wasmtime-wasi-http upstream.

Reply:
The least surprising behavior would probably be:

1. Default to just standard roots
2. If ca file is given, disable standard roots
3. Opt-in to combining the two above

Maybe something like:

```toml
ca_roots_file = "..."
ca_roots_webpki = true # defaults to true iff ca_roots_file not specified
```

Reply:
great. done.